jobs/bbr-credhubdb/spec

```diff
@@ -39,3 +39,6 @@ properties:
     default: true
   credhub.data_storage.tls_ca:
     description: "CA trusted for making TLS connections to targeted database server"
+  credhub.data_storage.hostname_verification.enabled:
+    description: "Toggle certificate host name verification. WARNING: Only use if your database does not include a matching hostname in their server certificate!"
+    default: false
```

Comment:
I was expecting the default value to be `true`, as it is the safer option. Is there a reason you set it to `false`? Is the current behavior (before this change) to never verify the hostname?

Reply:
fixed!

Reply:
I see where the confusion is arrising RE the true / false values.

BBR's internal key is called: skip_host_verify which should be FALSE by default as it's kind of the opposite behaviour to the credhub naming.

Within TAS, the property is already exposed here:

https://github.com/pivotal/tas/blob/a80f4cc7269ac8dd9c2033ce011951cc2971e3b6/tas/properties/credhub_database.yml#L39-L42

As you can see it's called `disable_hostname_verification`. But is then interpolated here: https://github.com/pivotal/tas/blob/a80f4cc7269ac8dd9c2033ce011951cc2971e3b6/tas/properties/credhub_database.yml#L54-L55

By default in TAS it's false. Which is ACTUALLY the inverse, which is super confusing. You are right, it should be true as default

Reply:
I have updated the PR to reflect that credhubs spec is:

hostname_verification.enabled ==> true by default

and BBRs is now:

skip_host_verify ==> false by default. (don't skip... therefore do verify)

They are the inverse of each other, and this now maintains the same behaviour internally.

Reply:
I see. I always try to convince people not to use negative words like "disable" in variable names or configuration fields, but I am not very successful :).

This kind of problem is usually we we like to test and TDD, but I do understand how difficult it is a write a test for this particular feature.

Looks like it's all good now, glad we fixed it.